websocket/_app.py

```diff
@@ -96,14 +105,18 @@ def __init__(self, app, ping_timeout, dispatcher):
         self.app = app
         self.ping_timeout = ping_timeout
         self.dispatcher = dispatcher
+        dispatcher.signal(2, dispatcher.abort) # keyboard interrupt
 
     def read(self, sock, read_callback, check_callback):
         self.dispatcher.read(sock, read_callback)
-        self.ping_timeout and self.dispatcher.timeout(self.ping_timeout, check_callback)
+        self.ping_timeout and self.timeout(self.ping_timeout, check_callback)
 
     def timeout(self, seconds, callback):
         self.dispatcher.timeout(seconds, callback)
```

Comment:
Should this be `self.timeout(seconds, callback)` instead of `self.dispatcher.timeout(seconds, callback)`?

Reply:
Yes - it's defined on line 114